File: ios/Samples/Showcase-objc/LiferayScreens-Showcase-Objc/DDLFormScreenletViewController.m
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
#import "DDLFormScreenletViewController.h"

@interface DDLFormScreenletViewController ()

@property (nonatomic, weak) IBOutlet DDLFormScreenlet* screenlet;
@property (nonatomic, weak) IBOutlet UITextField* recordIdField;

@end


@implementation DDLFormScreenletViewController

- (IBAction)loadForm:(id)sender {
	long long recordId = [self.recordIdField.text longLongValue];
	if (recordId != 0) {
		self.screenlet.recordId = recordId;
		[self.recordIdField resignFirstResponder];
		[self.screenlet loadRecord];
	}
	else {
		[self.screenlet loadForm];
	}
}

- (void)viewDidLoad {
	[super viewDidLoad];

	self.screenlet.delegate = self;
}

- (void)onFormLoaded:(DDLRecord *)record {
	NSLog(@"DELEGATE onFormLoaded -> %@", record);
}

- (void)onFormLoadError:(NSError *)error {
	NSLog(@"DELEGATE onFormLoadError -> %@", error);
}

- (void)onRecordLoaded:(DDLRecord *)record {
	NSLog(@"DELEGATE onRecordLoaded -> %@", record);
}

- (void)onRecordLoadError:(NSError *)error {
	NSLog(@"DELEGATE onRecordLoadError -> %@", error);
}

- (void)onFormSubmitted:(DDLRecord *)record {
	NSLog(@"DELEGATE onFormSubmitted -> %@", record);
}

- (void)onFormSubmitError:(NSError *)error {
	NSLog(@"DELEGATE onFormSubmitError -> %@", error);
}

- (void)onDocumentUploadStarted:(DDLFieldDocument *)field {
	NSLog(@"DELEGATE onDocumentUploadStarted -> %@", field);
}

- (void)onDocumentUploadedBytes:(DDLFieldDocument *)field bytes:(NSUInteger)bytes sent:(int64_t)sent total:(int64_t)total {
	NSLog(@"DELEGATE onDocumentUploadedBytes -> field=%@ %lu/%lld/%lld", field, (unsigned long)bytes, sent, total);
}

- (void)onDocumentUploadCompleted:(DDLFieldDocument *)field result:(NSDictionary *)result {
	NSLog(@"DELEGATE onDocumentUploadCompleted -> field=%@ result=%@", field, result);
}

- (void)onDocumentUploadError:(DDLFieldDocument *)field error:(NSError *)error {
	NSLog(@"DELEGATE onDocumentUploadError -> field=%@ error=%@", field, error);
}

@end

File: ios/Samples/Showcase-swift/LiferayScreens-Showcase-Swift/ViewControllers/DDL/DDLFormScreenletViewController.swift
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
import UIKit
import LiferayScreens


class DDLFormScreenletViewController: UIViewController, DDLFormScreenletDelegate {

	@IBOutlet weak var loadButton: UIButton? {
		didSet {
			loadButton?.replaceAttributedTitle(NSLocalizedString("load-button", comment: "LOAD"),
			                                   forState: .normal)
		}
	}
	@IBOutlet var screenlet: DDLFormScreenlet? {
		didSet {
			screenlet?.delegate = self
			screenlet?.recordSetId = LiferayServerContext.longPropertyForKey("ddlRecordSetId")
			screenlet?.structureId = LiferayServerContext.longPropertyForKey("ddlStructureId")
		}
	}
	@IBOutlet var recordIdTextField: UITextField? {
		didSet {
			recordIdTextField?.text =
				String(LiferayServerContext.longPropertyForKey("ddlRecordId"))
		}
	}

	@IBAction func loadForm(_ sender: AnyObject) {
		if let id = recordIdTextField?.text, let recordId = Int64(id) {
			screenlet?.recordId = recordId
			recordIdTextField?.resignFirstResponder()
			screenlet?.loadRecord()
		}
		else {
			screenlet?.loadForm()
		}
	}


	//MARK: DDLFormScreenletDelegate

	func screenlet(_ screenlet: DDLFormScreenlet, onFormLoaded record: DDLRecord) {
		LiferayLogger.logDelegateMessage(args: record)
	}
	
	func screenlet(_ screenlet: DDLFormScreenlet, onFormLoadError error: NSError) {
		LiferayLogger.logDelegateMessage(args: error)
	}
	
	func screenlet(_ screenlet: DDLFormScreenlet, onFormSubmitError error: NSError) {
		LiferayLogger.logDelegateMessage(args: error)
	}
	
	func screenlet(_ screenlet: DDLFormScreenlet, onRecordLoaded record: DDLRecord) {
		LiferayLogger.logDelegateMessage(args: record)
	}
	
	func screenlet(_ screenlet: DDLFormScreenlet, onRecordLoadError error: NSError) {
		LiferayLogger.logDelegateMessage(args: error)
	}
	
	func screenlet(_ screenlet: DDLFormScreenlet, onFormSubmitted record: DDLRecord) {
		LiferayLogger.logDelegateMessage(args: record)
	}
	
	func screenlet(_ screenlet: DDLFormScreenlet,
	               onDocumentFieldUploadStarted field: DDMFieldDocument) {
		LiferayLogger.logDelegateMessage(args: field)
	}
	
	func screenlet(_ screenlet: DDLFormScreenlet,
	               onDocumentField field: DDMFieldDocument,
	               uploadError error: NSError) {
		LiferayLogger.logDelegateMessage(args: error)
	}
	
	func screenlet(_ screenlet: DDLFormScreenlet,
	               onDocumentField field: DDMFieldDocument,
	               uploadResult result: [String : Any]) {
		LiferayLogger.logDelegateMessage(args: field, result)
	}
	
	func screenlet(_ screenlet: DDLFormScreenlet,
	               onDocumentField field: DDMFieldDocument,
				   uploadedBytes bytes: UInt64,
				   totalBytes total: UInt64) {
		LiferayLogger.logDelegateMessage(args: field)
	}

}
